From f61dd9c5324d8439c52764c7aebaac7d0d4b7128 Mon Sep 17 00:00:00 2001
From: David Seifert <soap@gentoo.org>
Date: Sat, 21 Dec 2019 14:02:13 +0100
Subject: [PATCH] Fix missing `boost/serialization/split_member.hpp` include

* As suggested by @glenfe and @robertramey in
  https://github.com/boostorg/serialization/commit/c32a663c9963385430abc563f9c85f94d8da43a9#r36528430,
  the `boost/serialization/split_member.hpp` include is not part
  of the interface and consumers should not rely on it being
  included.

Fixes lightspark/lightspark#406
Fixes https://bugs.gentoo.org/703294
---
 include/boost/bimap/detail/map_view_iterator.hpp | 1 +
 include/boost/bimap/detail/set_view_iterator.hpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/boost/bimap/detail/map_view_iterator.hpp b/include/boost/bimap/detail/map_view_iterator.hpp
index 17d43cb..4af8484 100644
--- a/include/boost/bimap/detail/map_view_iterator.hpp
+++ b/include/boost/bimap/detail/map_view_iterator.hpp
@@ -22,6 +22,7 @@
 
 #ifndef BOOST_BIMAP_DISABLE_SERIALIZATION
   #include <boost/serialization/nvp.hpp>
+  #include <boost/serialization/split_member.hpp>
 #endif // BOOST_BIMAP_DISABLE_SERIALIZATION
 
 #include <boost/iterator/detail/enable_if.hpp>
diff --git a/include/boost/bimap/detail/set_view_iterator.hpp b/include/boost/bimap/detail/set_view_iterator.hpp
index e785ed7..b1e2dfa 100644
--- a/include/boost/bimap/detail/set_view_iterator.hpp
+++ b/include/boost/bimap/detail/set_view_iterator.hpp
@@ -22,6 +22,7 @@
 
 #ifndef BOOST_BIMAP_DISABLE_SERIALIZATION 
   #include <boost/serialization/nvp.hpp>
+  #include <boost/serialization/split_member.hpp>
 #endif // BOOST_BIMAP_DISABLE_SERIALIZATION
 
 #include <boost/iterator/detail/enable_if.hpp>
